### ISSUE TITLE
CRUD fields that involve related entities show them using the default locale, ignoring the edit locale

### ISSUE THREAD

**Issue by dandarie** [Laravel-Backpack/CRUD#3228]
# Bug report

### What I did

I added a select2 field showing an attribute of a related entity which has multilanguage fields, including the displayed attribute.

```
Crud::addField([
     'name' => 'parent_id',
     'label' => 'Parent',
     'type' => 'select2',
     'attribute' => 'title'
]);
```

### What I expected to happen

The related entry attribute (parent title) should have been displayed in the current edit lanuage (derived from the locale GET parameter).

### What happened

When editing the entry in a language other than the default, the related entry attribute was displayed using the current app language.

This affects all fields that use related entries, I believe.

### What I've already tried to fix it

I temporarily set the app language according to the GET parameter, then set it back.

```
    @include('crud::fields.inc.translatable_icon')
    @php
        // get the current app locale
        $locale = app()->getLocale();
        // set the locale according to the GET param (or current, if missing)
        app()->setLocale(request()->get('locale') ?? $locale);
    @endphp
    <select
        name="{{ $field['name'] }}"
        ...
    </select>
    @php
        // set the locale value back to the current app locale
        app()->setLocale($locale)
    @endphp
```

However, it is unclear what side effects this fix could have.

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:

### PHP VERSION:
PHP 7.4.4 (cli) (built: Mar 20 2020 13:47:45) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.4, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
v8.6.0@a71952a6dba55de0bb11b5fbbd84874eda2a755c

### BACKPACK VERSION:
4.1.25@23fb1ed5153d5465d49d2fe1ca265a5985c7887

**Comment by pxpm** [comment]
Hello @dandarie 

Just to clarify, you are editing an entry let's say in French and your app locale is English. 

When using a field that fetch an related entry, the chosen attribute to display picks the English translation and not the French you are editing ?

If this is the case, I see this as a good feature.

Having a key like: 

```php
'translatable_attribute' => true //force the attibute to either use the editing lang -> app lang -> app fallback -> first available lang
```
By default false to keep it as is.

I think this would require changes in some field files, mainly the selects and relationships.

I tagged this accordingly.

Best,
Pedro

**Comment by dandarie** [comment]
I agree, it should be optional.

Furthermore, the `translatable_attribute` should be either `true` (to use the editing locale), a specific locale (such as 'fr'), or a closure (a user preferred language, perhaps).

**Comment by tabacitu** [comment]
Just talked to Pedro, and it sounds like the solution outlined by @promatik here - https://github.com/Laravel-Backpack/CRUD/pull/2253#discussion_r515920726

Basically:
- change the app_locale before the Update form
- change the app_locale after the Update form (actually just before the buttons)

This should theoretically be a perfect solution for this, and because the spatie package will pick up the dropdown options in their translated form. We'll also probably need to do this inside the Fetch operation, to cover AJAX requests too.

We're going to prototype this solution and see how it goes 👍 We'll keep you posted.

**Comment by pxpm** [comment]
We found other issues that the simplest solution of changing app locales before the form would not work that well. 

One of the major drawbacks is the fact that field labels for example also get translated. And that's not the point here, if that were the case, just change the `Backpack panel language`. 

This is postponed to 4.2 IMO. 

Best,
Pedro

**Comment by pxpm** [comment]
This has landed in 6.7.17 in https://github.com/Laravel-Backpack/CRUD/pull/4267

Closing up the issue. Please re-open if needed 🙏
